src/schoology-extractor/schoology_extractor/main.py

```diff
@@ -3,76 +3,128 @@
 # The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 # See the LICENSE and NOTICES files in the project root for more information.
 
+import logging
 import os
+from typing import Any, List
+import sys
 
 from dotenv import load_dotenv
 
 from helpers import export_data
 from api.request_client import RequestClient
 
+
 load_dotenv()
+# Configure logging
+log_level = os.getenv("SCHOOLOGY_LOG_LEVEL", "INFO")
+assert log_level in ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], "The specified `SCHOOLOGY_LOG_LEVEL` is not valid"
+logFormatter = '%(asctime)s - %(levelname)s - %(message)s'
+
+logging.basicConfig(
+    handlers=[
+        logging.StreamHandler(sys.stdout),
+    ],
+    format=logFormatter,
+    level=log_level
+)
+logger = logging.getLogger(__name__)
+
+
+logger.info("Starting Ed-Fi LMS Schoology Extractor")
+logger.debug("Loading and processing .env variables")
+
 schoology_key = os.getenv("SCHOOLOGY_KEY")
 schoology_secret = os.getenv("SCHOOLOGY_SECRET")
 schoology_output_path = os.getenv("SCHOOLOGY_OUTPUT_PATH")
 schoology_grading_periods = os.getenv("SCHOOLOGY_GRADING_PERIODS")
 
-assert schoology_key is not None, "A `SCHOOLOGY_KEY` must be present in the .env file and it was not found."
-assert schoology_secret is not None, "A `SCHOOLOGY_SECRET` must be present in the .env file and it was not found."
-assert schoology_output_path is not None, "A `SCHOOLOGY_OUTPUT_PATH` must be present in the .env file and it was not found."
-assert schoology_grading_periods is not None, "A `SCHOOLOGY_GRADING_PERIODS` must be present in the .env file and it was not found."
+assert (
+    schoology_key is not None
+), "A `SCHOOLOGY_KEY` must be present in the .env file and it was not found."
+assert (
+    schoology_secret is not None
+), "A `SCHOOLOGY_SECRET` must be present in the .env file and it was not found."
+assert (
+    schoology_output_path is not None
+), "A `SCHOOLOGY_OUTPUT_PATH` must be present in the .env file and it was not found."
+assert (
+    schoology_grading_periods is not None
+), "A `SCHOOLOGY_GRADING_PERIODS` must be present in the .env file and it was not found."
+
+
+grading_periods_array = schoology_grading_periods.split(",")
 
+logger.debug("Finished loading and processing .env variables")
```

Comment:
FYI, I do not usually write "finished..." type statements unless there is a strong need to look at the amount of time that a process takes.

Reply:
Removed